projects/commudle-admin/src/app/feature-modules/users/components/public-profile/user-basic-details/email-preferences/email-preferences.component.ts

```diff
@@ -0,0 +1,74 @@
+import { ChangeDetectorRef, Component } from '@angular/core';
+import { EmailUnsubscribeGroupsService } from 'projects/commudle-admin/src/app/feature-modules/email-confirmations/services/email-unsubscribe-groups.service';
+import { AppUsersService } from 'projects/commudle-admin/src/app/services/app-users.service';
+import { Subscription } from 'rxjs';
+
+@Component({
+  selector: 'app-email-preferences',
+  templateUrl: './email-preferences.component.html',
+  styleUrls: ['./email-preferences.component.scss'],
+})
+export class EmailPreferencesComponent {
+  subscriptionGroups: Object;
+  subscriptions: Subscription[] = [];
+  subscribeAll: boolean;
+
+  constructor(
+    private appUsersService: AppUsersService,
+    private emailUnsubscribeGroupsService: EmailUnsubscribeGroupsService,
+    private cdr: ChangeDetectorRef,
+  ) {}
+
+  ngOnInit(): void {
+    this.getAllUserSubscriptions();
+  }
+
+  ngOnDestroy(): void {
+    this.subscriptions.forEach((subscription) => subscription.unsubscribe());
+  }
+
+  getAllUserSubscriptions() {
+    this.subscriptions.push(
+      this.appUsersService.getUserEmailSubscriptions().subscribe((response) => {
+        this.subscriptionGroups = response;
+        this.checkAllSubscriptions();
+      }),
+    );
+  }
+
+  toggleSubscription(entity) {
+    this.subscriptions.push(
+      this.emailUnsubscribeGroupsService.toggleSubscription(entity.uuid).subscribe((response: boolean) => {
+        entity.subscribed = response;
+        this.checkAllSubscriptions();
+        this.cdr.detectChanges();
+      }),
+    );
+  }
+
+  checkAllSubscriptions() {
+    let status: boolean = false;
+    Object.keys(this.subscriptionGroups).forEach((group: string) => {
```

Comment:
Let's use `some` from lodash to make the code cleaner

Reply:
done